library/nxos_feature.py

```diff
@@ -0,0 +1,298 @@
+    returned: always
+    type: string
+    sample: "no feature eigrp ;"
+changed:
+    description: check to see if a change was made on the device
+    returned: always
+    type: boolean
+    sample: true
+'''
+
+import socket
+import xmltodict
+try:
+    HAS_PYCSCO = True
+    from pycsco.nxos.device import Device
+    from pycsco.nxos.device import Auth
+    from pycsco.nxos.error import CLIError
+except ImportError as ie:
+    HAS_PYCSCO = False
+
+
+def parsed_data_from_device(device, command, module):
+    try:
+        data = device.show(command)
+    except CLIError as clie:
+        module.fail_json(msg='Error sending {}'.format(command),
+                         error=str(clie))
+
+    data_dict = xmltodict.parse(data[1])
+    body = data_dict['ins_api']['outputs']['output']['body']
+
+    return body
+
+
+def apply_key_map(key_map, table):
+    new_dict = {}
+    for key, value in table.items():
+        new_key = key_map.get(key)
+        if new_key:
+            new_dict[new_key] = str(value)
+    return new_dict
+
+
+def apply_value_map(value_map, resource):
+    for key, value in value_map.items():
+        resource[key] = value[resource.get(key)]
+    return resource
+
+
+def command_list_to_string(command_list):
+    """Converts list of commands into proper string for NX-API
+
+    Args:
+        cmds (list): ordered list of commands
+
+    Returns:
+        str: string of commands separated by " ; "
+
+    """
+    if command_list:
+        command = ' ; '.join(command_list)
+        return command + ' ;'
+    else:
+        return ''
+
+
+def get_feature_state(available_features, feature):
+    if 'enabled' in available_features[feature]:
+        existstate = 'enabled'
+    else:
+        existstate = 'disabled'
+    return existstate
+
+
+def get_features(device, feature, module):
+    available_features = {}
+    command = 'show feature'
+    body = parsed_data_from_device(device, command, module)
+    if body:
+        try:
+            feature_table = body['TABLE_cfcFeatureCtrlTable']['ROW_cfcFeatureCtrlTable']
+            key_map = {
+                    "cfcFeatureCtrlName2": "feature",
+                    "cfcFeatureCtrlOpStatus2": "status"
+                    }
+            if isinstance(feature_table, dict):
+                feature_table = [feature_table]
+
+            for single_feature in feature_table:
+                features = apply_key_map(key_map, single_feature)
+                parsed_feature = features['feature']
+                available_features[parsed_feature] = features['status']
+        except:
+            splitted_body = body.split('\n')
+            for element in splitted_body:
+                splitted_element = element.split()
+                available_features[splitted_element[0]] = splitted_element[2]
+    return available_features
+
+
+def get_commands(proposed, existing, state, feature):
+    commands = []
+    delta = set(proposed.iteritems()).difference(existing.iteritems())
```

Comment:
Since there is only single key/value pair, you may just want to do a simple `if proposed != existing:` check here, no?